### DISCUSSION TITLE
Missing Variables & Incorrect Decimal Place Settings

### DISCUSSION THREAD

**Discussion by FunofFlying** [mpaperno/MSFSTouchPortalPlugin#66] [Q&A] [answered]
Hi there, I have already been quite successful setting up new TP pages using this plugin, but it appears that some variable outputs, such as one for the Cessna 172 transponder code as an example, are not available in the GUI drop down boxes or in the State.ini file? Further, the format of some variable outputs such as the NAV1/2 active and standby values are shown to three decimal places and not two as per the real aircraft?

I have tried to alter these in the State.ini file, but to no avail. Could you please advise therefore:-

1. Can new variable outputs be added to the plugin by the user and if so, how?
2. Is the anyway for the user to alter the format of decimal places shown for some variable outputs?

Thanks and regards

Paul

**Comment by mpaperno** [comment, marked as answer]
Hi Paul,

There is actually a whole article about it in the Wiki:  
https://github.com/mpaperno/MSFSTouchPortalPlugin/wiki/Using-Custom-States-and-Simulator-Variables

Since you mentioned radio, you may also find this set of custom definitions interesting (but read the above first so you know what this is all about):
https://github.com/mpaperno/MSFSTouchPortalPlugin/wiki/assets/pages/CustomRadioStates.ini
(more details here: https://github.com/mpaperno/MSFSTouchPortalPlugin/wiki/Pages-Buttons-and-Graphics#radio-stack-v2-4-x-7-layout-phone-size)

Cheers,
-Max

**Comment by FunofFlying** [comment]
Hi Max

Many thanks for your quick response, appreciated 😊

I have already looked through the excellent pages you have provided on this, but can’t find out how to create new output variables and/or alter output variable decimal place settings? Obviously I’ve tried amending the State.ini file, but no results so far.

However, I’ll have another look through the information already out there, but if I still can’t fathom it out, perhaps I could get back in touch?

Kind regards
Paul

**Reply by mpaperno** [reply]
The decimal places would be changed using the "Format" setting, or "StringFormat" in the INI files. You can limit to 2 decimals using "F2" as the format string.

Do not modify the included States.ini file, or at least realize that it will be replaced on the next plugin update.  This is mentioned in the article, where it talks about using INI files.

Of course feel free to come back with any other questions.

Cheers,
-Max

**Comment by FunofFlying** [comment]
Hi Max

Yes, thanks for the clarification about the State.ini file, as it happens, I’ve actually just read that part on your web page lol!

I’ll have another look in the file to see what I can do there? 

Initially I was altering the string format setting for the NAV1 active and standby variables in the State.ini file from 0.000# to 0.00#, but now I understand that the State.ini file doesn’t control the plugin, it’s a result of its output! 

Thanks again and take care

Paul 😊

**Reply by mpaperno** [reply]
"0.00#" means 2 _or_ 3 decimal places if there is another.  In the original it will show up to 4 if they're there.  A format like "F2" is explicit, or "0.00".

> the State.ini file doesn’t control the plugin, it’s a result of its output!

No actually the plugin does read that file to populate all the build-in state variables. The file is static and distributed with the plugin installation.  So it will get replaced next time you update the plugin.

**Comment by FunofFlying** [comment]
Forgive me Max, I actually learned how to write bespoke Arduino sketch codes for my YouTube channel on the subject of home flight simulators, so I’ll get there with this eventually!

 

As an exercise, I put the below lines of code in the State.ini file, so should I see this “Transponder Code” output variable as an available option in a drop down box when working in Touch Portal? So far, I can’t see this anywhere?

 

[TransponderSquawkCode]

CategoryId = Communication

Name = "Transponder squawk code"

SimVarName = "TRANSPONDER CODE:1"

Unit = "BCD16"

StringFormat = "0.000#"

DeltaEpsilon = 0.0009

 

Also as indicated before, I changed the NAV1 active and standby string format code from 0.000# to 0.00# in the State.ini file, but in Touch Portal, the variable still displayed three decimal places??

 

Thanks for your help my friend 😊

**Reply by mpaperno** [reply]
Uh...  there's an example of a working transponder code request in the file I linked earlier:

```ini
[TransponderCode]
CategoryId = Communication
Name = "Transponder 1 Code"
SimVarName = "TRANSPONDER CODE:1"
Unit = "number"
StringFormat = "0000"
```

> Also as indicated before, I changed the NAV1 active and standby string format code from 0.000# to 0.00# in the State.ini file, but in Touch Portal, the variable still displayed three decimal places??

Sorry, but didn't I just answer this question?  In the reply you quoted?

To be clear, after you make changes in any INI files they need to be re-loaded (the plugin provides actions for that as described in the article). The plugin doesn't monitor any files for changes.

HTH,
-Max

**Comment by FunofFlying** [comment]
Hi Max

Apologies, I’ve been staring at this stuff all week and at my age of 67, I think I’m getting a bit punch drunk lol!


I think it’s your last comment that I didn’t appreciate, in that you have to re-load the .ini file for any changes to become effective, if I understand you correctly?


Thanks again and I’ll now just crack on and try not to bother you again with silly questions 😊

Kind regards Paul
